src/shared/context/SelectedCartProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import { createContext, useState } from 'react';
+import { CartItem } from '../types/cart';
+
+interface SelectedCartContextType {
+  selectedCartItems: CartItem[];
+  updateSelectedCartItem: (item: CartItem, updatedQuantity: number) => void;
+  addAllCartItemsInSelected: (items: CartItem[]) => void;
+  removeSelectedCartItem: (item: CartItem) => void;
+}
+
+export const SelectedCartContext = createContext<SelectedCartContextType | undefined>(undefined);
+
+interface SelectedCartProviderProps {
+  children: React.ReactNode;
+}
+
+export const SelectedCartProvider = ({ children }: SelectedCartProviderProps) => {
+  const [selectedCartItems, setSelectedCartItems] = useState<CartItem[]>([]);
+
+  const updateSelectedCartItem = (cartItem: CartItem, quantity?: number) => {
+    setSelectedCartItems((prevItems) => {
+      const existingItemIndex = prevItems.findIndex((item) => item.id === cartItem.id);
+
+      if (existingItemIndex > -1) {
+        const updatedItems = [...prevItems];
+        updatedItems[existingItemIndex].quantity = quantity ?? 1;
+        return updatedItems;
+      }
```

Comment:
```suggestion
      const exists = prevItems.some(item => item.id === cartItem.id)

      if (exists) {
        return prevItems.map(item =>
          item.id === cartItem.id
            ? { ...item, quantity: updatedQuantity ?? 1 }
            : item
        )
      }
```
요기도 간단하게 some으로 요렇게 구현하는 건 어떠세요?

Reply:
@JUDONGHYEOK 
해당 피드백 감사합니다!
코드 효율성에 대해서 같이 얘기하고 싶습니다!
제가 구현한 코드에서는

```tsx
  const updateSelectedCartItem = (cartItem: CartItem, quantity?: number) => {
    setSelectedCartItems((prevItems) => {
      const existingItemIndex = prevItems.findIndex((item) => item.id === cartItem.id);

      if (existingItemIndex > -1) {
        const updatedItems = [...prevItems];
        updatedItems[existingItemIndex].quantity = quantity ?? 1;
        return updatedItems;
      }

      return [...prevItems, cartItem];
    });
  };

```

`findIndex` 메소드 통해서 한 번만 순회하면 해당 idnex를 이용해서 해당 로직을 처리할 수 있지만

`some` 메소드를 사용할 경우

```tsx
  const updateSelectedCartItem = (cartItem: CartItem, quantity?: number) => {
    setSelectedCartItems((prevItems) => {
      const existing = prevItems.some((item) => item.id === cartItem.id);

      if (existing) {
        return prevItems.map((item) =>
          item.id === cartItem.id ? { ...item, quantity: quantity !== undefined ? quantity : item.quantity } : item
        );
      }

      return [...prevItems, cartItem];
    });
  };

```

`some`메소드와 `map` 메소드를 사용하여 두 번을 순회해야할 것 같아서 해당 부분에서는 `findIndex`가 성능 측면에서 현재 상황에서는 미미할 수도 있겠지만 더 효율적이라고 생각하는데 혹시 동키콩은 어떻게 생각하시나요!

Reply:
오 좋은 관점이네요! 하지만 말씀하신대로 성능상의 이점이 그렇게 큰가를 고민해볼 필요는 있을 것 같아요. 현재 보여주신 코드는 어떻게 할지 설명하는 명령적인 방식이라 코드의 가독성이 떨어진다고 생각했었거든요. 언제나 트레이드오프는 존재하니 적절하게 고려하면 좋을 것 같습니다! 이부분은 그렇구나 하고 넘어가셔도 될것 같아요!😄

Reply:
피드백 주셔서 감사합니다! 동키콩의 의견을 들어보니 해당 코드를 다양한 시각으로 생각할 수 있게 되서 흥미롭네요 😮